yogstation/code/modules/mob/living/simple_animal/borer/borer_powers.dm

```diff
@@ -0,0 +1,463 @@
+		return
+
+	if(docile)
+		to_chat(src, "<span class='warning'>You are feeling far too docile to do that.</span>")
+		return
+
+	if(chemicals < 75)
+		to_chat(src, "<span class='warning'>You need 75 chems to punish your host.</span>")
+		return
+
+	if(!checkStrength())
+		return 0
+
+	var/punishment = input("Select a punishment:.", "Punish") as null|anything in list("Blindness","Deafness","Stun")
+
+	if(!punishment)
+		return
+
+	if(chemicals < 75)
+		to_chat(src, "<span class='warning'>You need 75 chems to punish your host.</span>")
+		return
+
+	switch(punishment) //Hardcoding this stuff.
+		if("Blindness")
+			victim.blind_eyes(2)
+		if("Deafness")
+			var/obj/item/organ/ears/ears = victim.getorganslot(ORGAN_SLOT_EARS)
+			ears.deaf = 20
+		if("Stun")
+			victim.Paralyze(10)
+
+	log_game("[src]/([src.ckey]) punished [victim]/([victim.ckey] with [punishment]")
+
+	chemicals -= 75
+
+
+mob/living/carbon/proc/release_control()
+
+	set category = "Borer"
+	set name = "Release Control"
+	set desc = "Release control of your host's body."
+
+	if(borer && borer.host_brain)
+		if(!borer.docile) //if the borer is docile they have to give up control regardless
+			if(alert(src, "Sure you want to give up your control so soon?", "Confirm", "Yes", "No") != "Yes")
+				return
+		to_chat(src, "<span class='danger'>You withdraw your probosci, releasing control of [borer.host_brain]</span>")
+
+		borer.detatch()
+
+		verbs -= /mob/living/carbon/proc/release_control
+		verbs -= /mob/living/carbon/proc/spawn_larvae
+
+/mob/living/carbon/proc/spawn_larvae()
+	set category = "Borer"
+	set name = "Reproduce"
+	set desc = "Vomit out your younglings."
+	var/poll_time = 100
+
+	if(isbrain(src))
+		to_chat(src, "<span class='usernotice'>You need a mouth to be able to do this.</span>")
+		return
+
+	if(!borer)
+		return
+
+	if(world.time < borer.next_spawn_time)
+		to_chat(src, "<span class='warning'>We are already reproducing.</span>")
+		return
+
+	if(borer.chemicals >= 100)
+		to_chat(src, "<span class='notice'>We prepare our host for the creation of a new borer.</span>")
+		borer.next_spawn_time = world.time + poll_time
+		var/list/Bcandidates = pollCandidates("Do you want to play as a borer?", ROLE_BORER, null, ROLE_BORER, poll_time) // we use this to FIND candidates. not necessarily use them.
+		if(!Bcandidates.len)
+			to_chat(src, "<span class='usernotice'>Our reproduction system seems to have failed... Perhaps we should try again some other time?</span>")
+			return
+
+		borer.chemicals -= 100
+
+		new /obj/effect/decal/cleanable/vomit(get_turf(src))
+		playsound(loc, 'sound/effects/splat.ogg', 50, 1)
+		visible_message("<span class='danger'>[src] heaves violently, expelling a rush of vomit and a wriggling, sluglike creature!</span>")
+
+		log_game("[src]/([src.ckey]) has spawned a new borer via reproducing.")
+		var/mob/living/simple_animal/borer/newborer = new(get_turf(src))
+		var/mob/dead/observer/O = pick(Bcandidates)
+		newborer.key = O.key
```

Comment:
Use the newborer.transfer_personality(O.client) proc instead, since it was made for this.